scylla_phot_culls.py

```diff
@@ -0,0 +1,183 @@
+        number of filters, and the cuts are applied in order based on the filter order.
+    errcut : float or a list of floats
+        If single value, cut is applied to all filters. If multiple values, the number of values has to match the
+        number of filters, and the cuts are applied in order based on the filter order.
+    plot : Boolean
+        If True, generate two plots as a check and a visualization of the data with quality cuts:
+        1) spatial plot
+        2) CMDs: if more than one filter is provided, plot a CMD with color=filter1-filter2 and mag=filter1
+    """
+
+    field_id = catalog_in.split('.st')[0].split('/15891_')[1]
+    print('Culling catalog for %s ' % field_id)
+
+    if catalog_out is None:
+        catalog_out = catalog_in.replace('.st', '.vgst')
+
+    if errcut:
+        catalog_out = catalog_out.replace('.fits', '_errcut.fits')
+        print(catalog_out)
+        if len(errcut) < len(filters):
+            errcut = np.linspace(errcut, errcut, len(filters))
+
+    # turn single-valued into multi-valued arrays for proper referencing in the for loop below
+    if len(crowdcut) < len(filters):
+        crowdcut = np.linspace(crowdcut, crowdcut, len(filters))
+    if len(sharpcut) < len(filters):
+        sharpcut = np.linspace(sharpcut, sharpcut, len(filters))
+    if len(roundcut) < len(filters):
+        roundcut = np.linspace(roundcut, roundcut, len(filters))
+
+    # Create a copy of the catalog file to work with
+    os.system("cp " + catalog_in + " " + catalog_out)
+
+    t = Table.read(catalog_out)
+    start_time_cull = time.time()
+
+    # Set the following entries outside the desired cut range to 99.999, or 99 ('FLAG')
+    # VEGA, STD, ERR, CHI, SNR, SHARP, ROUND, CROWD, FLAG
+    for i in range(len(filters)):
+        if filters[i] == 'F336W':
+            continue
+        inds_to_cut = np.where((t[filters[i] + '_CROWD'] > crowdcut[i]) |
+                               (t[filters[i] + '_SHARP'] * t[filters[i] + '_SHARP'] > sharpcut[i]) |
+                               (t[filters[i] + '_ROUND'] > roundcut[i]))[0]
+        if errcut:
+            errinds_to_cut = np.where(t[filters[i] + '_ERR'] > errcut)[0]
+            inds_to_cut = np.unique(np.concatenate((inds_to_cut, errinds_to_cut), 0))
+
+        t[filters[i] + '_VEGA'][inds_to_cut] = 99.999
+        t[filters[i] + '_STD'][inds_to_cut] = 99.999
+        t[filters[i] + '_ERR'][inds_to_cut] = 99.999
+        t[filters[i] + '_CHI'][inds_to_cut] = 99.999
+        t[filters[i] + '_SNR'][inds_to_cut] = 99.999
+        t[filters[i] + '_SHARP'][inds_to_cut] = 99.999
+        t[filters[i] + '_ROUND'][inds_to_cut] = 99.999
+        t[filters[i] + '_CROWD'][inds_to_cut] = 99.999
+        t[filters[i] + '_FLAG'][inds_to_cut] = 99
+
+    t.write(catalog_out, overwrite=True)
+    print('Done culling in %.2f sec.' % round((time.time()-start_time_cull), 2))
+    print('New catalog:' % catalog_out)
```

Comment:
Change
`print('New catalog:' % catalog_out)
`
to
`
print('New catalog: %s' % catalog_out)
`